File: src/main/java/core/procedures/io/CurrentOutputPort.java
package core.procedures.io;

import core.Repl;
import core.procedures.AFn;
import core.procedures.FnArgsBuilder;

public final class CurrentOutputPort extends AFn {

  public CurrentOutputPort() {
    super(new FnArgsBuilder().max(0).build());
  }

  @Override
  public String getName() {
    return "current-output-port";
  }

  @Override
  public Object apply0() {
    return Repl.getCurrentOutputPort();
  }
}

File: src/main/java/core/procedures/io/CurrentOutputPort.kt
package core.procedures.io

import core.Repl
import core.procedures.AFn
import core.procedures.Arity.Exactly
import core.scm.OutputPort

class CurrentOutputPort : AFn<Nothing, OutputPort>(name = "current-output-port", arity = Exactly(0)) {

    override operator fun invoke() = Repl.currentOutputPort
}
